### ISSUE TITLE
Linux release tagged "Fedora" actually does not work on Fedora 30

### ISSUE THREAD

**Issue by truenicoco** [medInria/medInria-public#495]
Title pretty much says it all. Application launch correctly, but fails to open .mha file. But the good new is: the release tagged Ubuntu actually works without problems with Fedora. Does it make sense to ship 2 different versions in this context?

Logs when trying (and failing) to open an image:

```
INFO  - Wed Nov 20 14:59:21 2019 - #################################### 
INFO  - Wed Nov 20 14:59:21 2019 - Version:  3.1.0 
INFO  - Wed Nov 20 14:59:21 2019 - Build Date:  20_11_2019 
WARN  - Wed Nov 20 14:59:21 2019 - "Unable to load /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins/libanimaMCMEstimationPlugin.so - Cannot load library /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins/libanimaMCMEstimationPlugin.so: (/home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins/libanimaMCMEstimationPlugin.so: undefined symbol: _ZTIN5anima24BaseIsotropicCompartmentE)" 
WARN  - Wed Nov 20 14:59:22 2019 - "Your config does not seem to be set correctly." 
WARN  - Wed Nov 20 14:59:22 2019 - "Please set  \"MEDINRIA_PLUGINS_DIR_LEGACY\"" 
WARN  - Wed Nov 20 14:59:22 2019 - "Default directory would be:  \"/home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy\"" 
WARN  - Wed Nov 20 14:59:22 2019 - "Actual directory is:  \"/home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/plugins_legacy:/home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy:\"" 
WARN  - Wed Nov 20 14:59:22 2019 - Failed to load plugins from path  "/home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/plugins_legacy" . Could not cd to directory. 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "animaDenseBMRegistrationPlugin" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "animaPyramidalBMRegistrationPlugin" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "Diffeomorphic Demons" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "Iterative Closest Point" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "itkDataDiffusionGradientListPlugin" 
WARN  - Wed Nov 20 14:59:22 2019 - "Unable to load /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libitkDataImagePlugin.so - Cannot load library /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libitkDataImagePlugin.so: (libicuuc.so.57: cannot open shared object file: No such file or directory)" 
DEBUG - Wed Nov 20 14:59:22 2019 - add error message to pluginManager: 
DEBUG - Wed Nov 20 14:59:22 2019 - 	 "Unable to load /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libitkDataImagePlugin.so - Cannot load library /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libitkDataImagePlugin.so: (libicuuc.so.57: cannot open shared object file: No such file or directory)" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "itkDataSHImagePlugin" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "itkDataTensorImagePlugin" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "ITK Filters" 
WARN  - Wed Nov 20 14:59:22 2019 - "Unable to load /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libitkINRDataImageReaderPlugin.so - Cannot load library /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libitkINRDataImageReaderPlugin.so: (libicuuc.so.57: cannot open shared object file: No such file or directory)" 
DEBUG - Wed Nov 20 14:59:22 2019 - add error message to pluginManager: 
DEBUG - Wed Nov 20 14:59:22 2019 - 	 "Unable to load /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libitkINRDataImageReaderPlugin.so - Cannot load library /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libitkINRDataImageReaderPlugin.so: (libicuuc.so.57: cannot open shared object file: No such file or directory)" 
WARN  - Wed Nov 20 14:59:22 2019 - "Unable to load /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libitkINRDataImageWriterPlugin.so - Cannot load library /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libitkINRDataImageWriterPlugin.so: (libicuuc.so.57: cannot open shared object file: No such file or directory)" 
DEBUG - Wed Nov 20 14:59:22 2019 - add error message to pluginManager: 
DEBUG - Wed Nov 20 14:59:22 2019 - 	 "Unable to load /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libitkINRDataImageWriterPlugin.so - Cannot load library /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libitkINRDataImageWriterPlugin.so: (libicuuc.so.57: cannot open shared object file: No such file or directory)" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "itkProcessRegistrationILogDemonsPlugin" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "LCCLogDemonsPlugin" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "Manual Registration" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "Algorithm Paint" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "Binary Operation" 
DEBUG - Wed Nov 20 14:59:22 2019 -  Loading plugin :  "Create Mesh From Mask" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "Filtering Workspace" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "Mask Application" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "Meshing Workspace" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "N4 Bias Correction" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "Registration Workspace" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "Remeshing" 
WARN  - Wed Nov 20 14:59:23 2019 - "Unable to load /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libmedSegmentationPlugin.so - Cannot load library /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libmedSegmentationPlugin.so: (libicuuc.so.57: cannot open shared object file: No such file or directory)" 
DEBUG - Wed Nov 20 14:59:23 2019 - add error message to pluginManager: 
DEBUG - Wed Nov 20 14:59:23 2019 - 	 "Unable to load /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libmedSegmentationPlugin.so - Cannot load library /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libmedSegmentationPlugin.so: (libicuuc.so.57: cannot open shared object file: No such file or directory)" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "Segmentation Workspace" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "medVtkFibersDataPlugin" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "medVtkViewPlugin" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "Mesh Manipulation" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "Mesh Mapping" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "Polygon ROI" 
WARN  - Wed Nov 20 14:59:23 2019 - "Unable to load /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libqtdcmDataSourcePlugin.so - Cannot load library /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libqtdcmDataSourcePlugin.so: (libicuuc.so.57: cannot open shared object file: No such file or directory)" 
DEBUG - Wed Nov 20 14:59:23 2019 - add error message to pluginManager: 
DEBUG - Wed Nov 20 14:59:23 2019 - 	 "Unable to load /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libqtdcmDataSourcePlugin.so - Cannot load library /home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins_legacy/libqtdcmDataSourcePlugin.so: (libicuuc.so.57: cannot open shared object file: No such file or directory)" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "qtshanoirDataSourcePlugin" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "Reformat" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "undoRedoRegistrationPlugin" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "Variational Segmentation" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "VOI Cutter" 
DEBUG - Wed Nov 20 14:59:23 2019 -  Loading plugin :  "vtkDataMeshPlugin" 
WARN  - Wed Nov 20 14:59:23 2019 - Unable to find a valid implementation of the store scp service. 
DEBUG - Wed Nov 20 14:59:23 2019 - factory creates dataSource: "qtshanoirDataSource" 
WARN  - Wed Nov 20 14:59:23 2019 - "libpng warning: iCCP: known incorrect sRGB profile" 
DEBUG - Wed Nov 20 14:59:23 2019 - no plugin path configured, use default: "/builds/work/nightly/medinria-nightly/src_ExtProjs/src/plugins/dtkDistributed" 
DEBUG - Wed Nov 20 14:59:23 2019 - initialize composer plugins using path: "/home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/plugins:/home/ncedilni/dl/medInria-3.0.1-Fedora_26-x86_64/bin/plugins:" 
INFO  - Wed Nov 20 14:59:25 2019 - ### Application is running... 
WARN  - Wed Nov 20 14:59:31 2019 - "Reader was unable to read at least:  \"/user/ncedilni/home/results/mihaela_atlas/circumferentials.mha\"" 
INFO  - Wed Nov 20 14:59:38 2019 - ### Application is closing... 
INFO  - Wed Nov 20 14:59:42 2019 - #################################### 
```

**Comment by ocommowi** [comment]
It sounds like you are missing a lib on your fedora. The itk data plugin which is needed to view your files cannot load because of that.

Do you have libicu installed ? It seems to be somewhat a dependency of some qt libs so it should be installed but if it's not there, you can try  yum install libicu (or something like that)

**Comment by truenicoco** [comment]
It is installed.

```
$ sudo dnf install libicu
[...]
Package libicu-63.2-2.fc30.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!
```

The binary tagged "Ubuntu" works just fine on the same machine, this is how I circumvented the problem. Some users may give up before me, so with this bug report I was just pointing that the download page is misleading. Fedora 30 users should download the "Ubuntu" version.

**Comment by ocommowi** [comment]
Yes but using the ubuntu version won't work on every fedora, so no that's not a good solution. It may be one for fedora 30 but not everyone.

In any case we have the reason at least. Fedora 30 is using libicu 63 when fedora 26 is using libicu 57. We may need to upgrade the minimum version required for fedora by upgrading our slaves os so that the problem solves itself.

**Comment by Florent2305** [comment]
After many tests and one correction the decisions is to publish a disclaimer about incompatibility between different version of fedora and tricks to try to run anyway.

**Comment by Florent2305** [comment]
Not closed but re-planned to milestones 4.1 in hope to find a better solutions.

**Comment by mathildemerle** [comment]
Solved Fedora 33
